### PR TITLE
add WorldSyncTimeEvent

### PR DESCRIPTION
**Pull request by BrandPVP**
Allows making per-player world time

### CODE REVIEW

File: src/world/World.php [hunk 1]
```diff
@@ -899,8 +900,12 @@ public function sendTime(Player ...$targets) : void{
 		if(count($targets) === 0){
 			$targets = $this->players;
 		}
+
 		foreach($targets as $player){
-			$player->getNetworkSession()->syncWorldTime($this->time);
+			$ev = new WorldSyncTimeEvent($this, $player, $this->time);
```
Comment (jasonw4331) [inline src/world/World.php]: It's preferred to hold an array of all recipients instead of spamming out event calls to all handlers

**Review by jasonw4331 — CHANGES_REQUESTED** [review]

File: src/world/World.php (hunk 1 above)
Comment (BrandPVP) [reply src/world/World.php]: That was in the first commit. But as it turns out, it doesn't affect performance so we can call it for every player

Comment (dktapps) [reply src/world/World.php]: The original way was preferable

Comment (BrandPVP) [reply src/world/World.php]: I'm not sure if the original way was that useful.. The only thing you can do with that is cancel the event and send a packet on your own. But we can add `hasHandlers()` check I guess?

Comment (dktapps) [reply src/world/World.php]: I don't like calling an event for every player when it's so easily avoided.

Comment (BrandPVP) [reply src/world/World.php]: What about now?

**Comment by dktapps** [comment]
I don't feel like this is the right way to support per-player time. Waiting for an event to do it just seems weird.

**Comment by dktapps** [comment]
Closing this as I don't think it's the right way to implement the feature you want.

**Pull request closed without merging**